src/panels/config/blueprint/input/types/ha-blueprint-input-input.ts

```diff
@@ -0,0 +1,332 @@
+                "ui.panel.config.blueprint.editor.inputs.type.single.icon"
+              ),
+            ],
+            [
+              "label",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.label"
+              ),
+            ],
+            [
+              "language",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.language"
+              ),
+            ],
+            [
+              "location",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.location"
+              ),
+            ],
+            [
+              "media",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.media"
+              ),
+            ],
+            [
+              "number",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.number"
+              ),
+            ],
+            [
+              "object",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.object"
+              ),
+            ],
+            [
+              "qr_code",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.qr_code"
+              ),
+            ],
+            [
+              "select",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.select"
+              ),
+            ],
+            [
+              "state",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.state"
+              ),
+            ],
+            [
+              "target",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.target"
+              ),
+            ],
+            [
+              "template",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.template"
+              ),
+            ],
+            [
+              "text",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.text"
+              ),
+            ],
+            [
+              "time",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.time"
+              ),
+            ],
+            [
+              "theme",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.theme"
+              ),
+            ],
+            [
+              "trigger",
+              localize(
+                "ui.panel.config.blueprint.editor.inputs.type.single.trigger"
+              ),
+            ],
+          ],
+        },
+        {
+          name: "default",
+          selector: defaultSelector,
+        },
+        // TODO: Sub-selector settings (https://www.home-assistant.io/docs/blueprint/selectors/)
```

Comment:
We have a `ha-selector-selector` already that allows for picking a selector type and configuring its options and defaults. Kind of feels like that's what you're trying to do here, maybe you can leverage that instead?

It's currently used in the script editor for `Add Fields`.

Reply:
Yes, that was exactly what I wanted! Thanks for letting me know about that. Just pushed an update - that helped cut down on a lot of code!